system/GlobalCompletionEvent.hpp

```diff
@@ -286,8 +327,23 @@ class GlobalCompletionEvent : public CompletionEvent {
       }
       DVLOG(3) << "fell thru conservative check";
     }
-    CHECK(!event_in_progress) << " gce(" << this << ", count:" << count << ")";
-    CHECK_EQ(count, 0);
+  }
+
+  /// Set up GCE to re-enroll this many cores/tasks on master core at completion.
+  /// This should be called on a single core after enrolling as many tasks as you want to reenroll on the master core (0).
```

Comment:
Technically this function can be called on any core, but the tasks have to be enrolled on the master core. :-) Yuck.

I thought about making this more complex: set it up so that every core can remember how many tasks are enrolled locally, and then re-enroll them on each core after the barrier. That would let you do more complex/unbalanced things (set one core up to expect a billion completions, and a handful of others to expect a few), but I can't think of an actual use case.